File: output/deepseek_deepseek_v3_2/zero/python_noimg/Arquivos.py

import sys

def main():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    N = int(data[0])
    B = int(data[1])
    files = list(map(int, data[2:2+N]))
    
    files.sort()
    
    left = 0
    right = N - 1
    folders = 0
    
    while left <= right:
        if left == right:
            folders += 1
            break
        if files[left] + files[right] <= B:
            left += 1
            right -= 1
        else:
            right -= 1
        folders += 1
    
    print(folders)

if __name__ == "__main__":
    main()

File: output/csbc_semish_2026/deepseek_deepseek_v3_2/few/cpp/Arquivos.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    long long b;
    if (!(cin >> n >> b)) return 0;

    vector<long long> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a.begin(), a.end());

    int left = 0, right = n - 1;
    int folders = 0;

    while (left <= right) {
        if (left == right) {
            folders++;
            break;
        }
        if (a[left] + a[right] <= b) {
            folders++;
            left++;
            right--;
        } else {
            folders++;
            right--;
        }
    }

    cout << folders << endl;
    return 0;
}